const Camion = require('../models/camion.model');

const getAllCamiones = async (req, res) => {
  try {
    const camiones = await Camion.obtenerTodos();
    res.json(camiones);
  } catch (error) {
    console.error('‚ùå Error al obtener camiones:', error);
    res.status(500).json({ message: 'Error al obtener camiones' });
  }
};

const createCamion = async (req, res) => {
  const { placa, capacidad_kg, estado_operativo, ubicacion_actual, km_actual } = req.body;

  try {
    const nuevoCamion = await Camion.crear({ placa, capacidad_kg, estado_operativo, ubicacion_actual, km_actual });
    res.status(201).json(nuevoCamion);
  } catch (error) {
    console.error('‚ùå Error al crear cami√≥n:', error);
    res.status(500).json({ message: 'Error al crear cami√≥n' });
  }
};


const updateCamion = async (req, res) => {
  const { id_camion } = req.params;
  const data = req.body; // ‚ö†Ô∏è Verifica si llega vac√≠o
  //log para ver los datos que llegan
  console.log('üì¶ Datos recibidos para actualizar:', req.body);


  try {
    const camionActualizado = await Camion.actualizar(id_camion, data);
    res.json(camionActualizado);
  } catch (error) {
    console.error('‚ùå Error al actualizar cami√≥n:', error);
    res.status(500).json({ message: 'Error al actualizar cami√≥n' });
  }
};


const deleteCamion = async (req, res) => {
  const { id_camion } = req.params;
  try {
    await Camion.eliminar(id_camion);
    res.json({ message: 'Cami√≥n eliminado correctamente' });
  } catch (error) {
    console.error('‚ùå Error al eliminar cami√≥n:', error);
    res.status(500).json({ message: 'Error al eliminar cami√≥n' });
  }
};

module.exports = {
  getAllCamiones,
  createCamion,
  updateCamion,
  deleteCamion,
};
